src/rocker/core.py

```diff
@@ -338,14 +347,27 @@ def write_files(extensions, args_dict, target_directory):
 
 def generate_dockerfile(extensions, args_dict, base_image):
     dockerfile_str = ''
+    # Preamble snippets
     for el in extensions:
         dockerfile_str += '# Preamble from extension [%s]\n' % el.name
         dockerfile_str += el.get_preamble(args_dict) + '\n'
     dockerfile_str += '\nFROM %s\n' % base_image
+    # ROOT snippets
     dockerfile_str += 'USER root\n'
     for el in extensions:
         dockerfile_str += '# Snippet from extension [%s]\n' % el.name
         dockerfile_str += el.get_snippet(args_dict) + '\n'
+    # Set USER if user extension activated
+    if 'user' in args_dict and args_dict['user']:
+        if 'user_override_name' in args_dict and args_dict['user_override_name']:
+            username = args_dict['user_override_name']
+        else:
+            username = get_user_name()
+        dockerfile_str += f'USER {username}\n'
```

Comment:
Should we consider adding `USER <user>` string in front of each extension call? Just in case the extension calls `USER root` in the user snippet? Or should that case be ignored since that would be non-standard

Reply:
That would be non-standard and also add a lot of potential layers to each build which is something that could be a limitation. It would be good to add docs that the snippets are expected not to change the USER.